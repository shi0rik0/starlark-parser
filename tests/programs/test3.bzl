s = "abcd"[::2]
a = [1, 2, 3, [1, 2, 3,],]
a[::]
a[0::]
a[:0:]
a[0:1:]
a[::0] 
a[0::2]    
a[:1:2]
a[0:1:2]



a[:]
a[1:]
a[:1]
a[1:10]

a[2]