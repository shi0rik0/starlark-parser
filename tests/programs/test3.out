[Assign] IDENTIFIER(s)=SLICE(STRING("abcd"),Slice(NONE():NONE():INT(2))
[Assign] IDENTIFIER(a)=LIST(INT(1),INT(2),INT(3),LIST(INT(1),INT(2),INT(3)))
[Expr] SLICE(IDENTIFIER(a),Slice(NONE():NONE():NONE())
[Expr] SLICE(IDENTIFIER(a),Slice(INT(0):NONE():NONE())
[Expr] SLICE(IDENTIFIER(a),Slice(NONE():INT(0):NONE())
[Expr] SLICE(IDENTIFIER(a),Slice(INT(0):INT(1):NONE())
[Expr] SLICE(IDENTIFIER(a),Slice(NONE():NONE():INT(0))
[Expr] SLICE(IDENTIFIER(a),Slice(INT(0):NONE():INT(2))
[Expr] SLICE(IDENTIFIER(a),Slice(NONE():INT(1):INT(2))
[Expr] SLICE(IDENTIFIER(a),Slice(INT(0):INT(1):INT(2))
[Expr] SLICE(IDENTIFIER(a),Slice(NONE():NONE())
[Expr] SLICE(IDENTIFIER(a),Slice(INT(1):NONE())
[Expr] SLICE(IDENTIFIER(a),Slice(NONE():INT(1))
[Expr] SLICE(IDENTIFIER(a),Slice(INT(1):INT(10))
[Expr] SLICE(IDENTIFIER(a),Slice(INT(2))
